File: cptr1/p3/main.go
package perfect

func isPerfect(n int) bool {

	if n <= 0 {
		return false
	}

	sum := 0
	for i := n - 1; i > 0; i-- {
		if n%i == 0 {
			sum += i
		}
	}

	return sum == n
}

File: cptr1/p3/main.mjs
const isPerfect = n => {
  if (n <= 0) {
    return false;
  }

  let sum = 0;
  for (let i = n - 1; i > 0; i--) {
    if (n % i === 0) {
      sum += i;
    }
  }
  return sum === n;
};

export default isPerfect;